src/context/fetcher/TenantBillingDetails.tsx

```diff
@@ -0,0 +1,69 @@
+import { MAX_TENANTS } from 'api/billing';
+import FullPageError from 'components/fullPage/Error';
+import { useUserInfoSummaryStore } from 'context/UserInfoSummary/useUserInfoSummaryStore';
+import { useTenantsDetailsForPayment } from 'hooks/useTenants';
+import { createContext, useContext } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useEntitiesStore_tenantsWithAdmin } from 'stores/Entities/hooks';
+import { BaseComponentProps, TenantPaymentDetails } from 'types';
+
+export interface TenantContextData {
+    tenantBillingDetails: TenantPaymentDetails[] | null;
+}
+const TenantContext = createContext<TenantContextData>({
+    tenantBillingDetails: null,
+});
+
+const TenantBillingDetailsContextProvider = ({
+    children,
+}: BaseComponentProps) => {
+    const hasSupportRole = useUserInfoSummaryStore(
+        (state) => state.hasSupportAccess
+    );
+
+    const tenantsWithAdmin = useEntitiesStore_tenantsWithAdmin();
+
+    // TODO (payment method notification)
+    // This does work a bit off of a side effect but that is fine for now.
+    //  Eventually I'd like to move this provider logic into the payment method
+    //  warning handled itself. As it is really the thing that care about these values.
+
+    // We suppress payment warnings for support role anyway so don't bother fetching for them
+    const { tenants, error, isValidating } = useTenantsDetailsForPayment(
+        hasSupportRole ? [] : tenantsWithAdmin.slice(0, MAX_TENANTS)
+    );
```

Comment:
Initially I thought about moving some of these hook calls into the payment fetch hook but decided against it. I could easily see needing that hook to look up payment details on a specific tenant to open a modal or something. So leaving out here for now.